tests/nat/agent/test_reasoning_agent.py

```
@@ -300,9 +302,11 @@ def mock_get_function(name: str):
 
     # The builder says we have 2 tools
     class FakeDeps:
-        functions = ["ToolA", "ToolB"]
+        functions = {"ToolA", "ToolB"}
```

Ruff (0.12.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)